helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -303,6 +303,53 @@ processors:
      - k8s.pod.name
      - k8s.pod.uid
   {{- end }}
+  {{- if .Values.splunkPlatform.fieldNameConvention.renameFieldsSck }}
+  resource/sckcompatible:
+    attributes:
+    - key: container_name
+      from_attribute: k8s.container.name
+      action: upsert
+    - key: cluster_name
+      from_attribute: k8s.cluster.name
+      action: upsert
+    - key: container_id
+      from_attribute: k8s.container.id
+      action: upsert
+    - key: pod
+      from_attribute: k8s.pod.name
+      action: upsert
+    - key: pod_uid
+      from_attribute: k8s.pod.uid
+      action: upsert
+    - key: namespace
+      from_attribute: k8s.namespace.name
+      action: upsert
+    {{- range $_, $label := .Values.extraAttributes.podLabels }}
+    - key: {{ printf "label_%s" $label }}
+      from_attribute: {{ printf "k8s.pod.labels.%s" $label }}
+      action: upsert
+    {{- end }}
+  {{- end }}
+  {{- if not .Values.splunkPlatform.fieldNameConvention.keepOtelContention }}
+  resource/removedups:
+    attributes:
+    - key: k8s.container.name
+      action: delete
+    - key: k8s.cluster.name
+      action: delete
+    - key: k8s.container.id
+      action: delete
+    - key: k8s.pod.name
+      action: delete
+    - key: k8s.pod.uid
+      action: delete
+    - key: k8s.namespace.name
+      action: delete
+    {{- range $_, $label := .Values.extraAttributes.podLabels }}
+    - key: {{ printf "k8s.pod.labels.%s" $label }}
+      action: delete
+    {{- end }}
```

Comment:
Do we need two more processors for this? Since it's applied to logs only, we could add these attribute manipulation rules in existing `resource/logs` helper. It should give better performance and no need to copy this logic across agent and gateway.

Also then you'd need to put `resource/logs` after `resource` processor because `resource` processor adds `k8s.cluster.name` that you want to rename.

Reply:
can even the `removedups` be in a single processor? it wouldn't mess up things by like deleting the attribute before inserting it?

Reply:
All the attribute actions are evaluated sequentially in resource processor. So just make sure they are put in the right order